spec/enum_spec.rb

```diff
@@ -0,0 +1,156 @@
+require '../enum.rb'
+
+RSpec.describe Enumerable do
+  let(:array1) { [0, 1, 2, 3, 4, 5, 6, 7, 8, 9] }
+  let(:array2) { %w[hello hey hi holla yea good] }
+  let(:array3) { [1, 2, 3, 4] }
+  let(:array4) { [2, 4, 6, 8] }
+  let(:array5) { [2, 4, 6, 8, 'jersey', %w[why me cr7 oa7]] }
+  let(:array6) { [4, 2, 3, 4, 4, 3, 5, 4, 4] }
+  let(:new_array) { [] }
+
+  describe '#my_each' do
+    it 'Passes each element to a new array based on some operation' do
+      array1.my_each { |ele| new_array << ele * 1 }
+
+      expect(new_array).to eql(array1)
+    end
+
+    it 'Returns a new array based on some arguments passed' do
+      array1.my_each { |ele| new_array << ele if ele.even? && ele > 0 }
+
+      expect(new_array).to eql(array4)
+    end
+
+    it 'Returns the Enumerator when no block is given' do
+      expect(array1.my_each).to be_kind_of(Enumerator)
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'Returns the Enumerator when no block is given' do
+      expect(array1.my_each_with_index).to be_kind_of(Enumerator)
+    end
+
+    it 'Returns element plus index when argument is passed' do
+      array3.my_each_with_index { |ele, index| new_array << "#{index} : #{ele}" }
+
+      expect(new_array).to eql(['0 : 1', '1 : 2', '2 : 3', '3 : 4'])
+    end
+
+    it 'Return the index position and element when specific results are wanted' do
+      array3.my_each_with_index { |ele, index| new_array << "#{index} : #{ele}" if ele.even? }
+
+      expect(new_array).to eql(['1 : 2', '3 : 4'])
+    end
+  end
+
+  describe '#my_select' do
+    it 'Returns specific data from the array based on the argument passed' do
+      array2.my_select { |ele| new_array << ele if ele.include?('h') }
+      expect(new_array).to eq(%w[hello hey hi holla])
+    end
```

Comment:
Please check the negative cases. When the array doesn't include the element you are looking for.